### PR TITLE
Adding test to check if OnDeserialized doesn't throw if serialization…

### PR DESCRIPTION
**Pull request by ViktorHofer**
… fields aren't initialized yet. Relates to https://github.com/dotnet/corefx/pull/21322

### CODE REVIEW

**Comment by ViktorHofer** [comment]
Unrelated failing tests in System.Drawing.Common. cc @hughbe

**Comment by hughbe** [comment]
Yeah I've seen these failures pop up today - it seems like 4 tests are failing on Windows 10 Release only across the board.

Opened #21360

File: src/System.ObjectModel/tests/ObservableCollection/ObservableCollection_Serialization.cs [hunk 1]
```diff
@@ -1,8 +1,10 @@
+
```
Comment (stephentoub) [inline src/System.ObjectModel/tests/ObservableCollection/ObservableCollection_Serialization.cs]: nit: Why the added blank line?

Comment (ViktorHofer) [inline src/System.ObjectModel/tests/ObservableCollection/ObservableCollection_Serialization.cs]: thx

**Pull request merged**